src/batou_ext/file.py

```diff
@@ -92,17 +93,26 @@ def remove(links, el):
 
     def _list_removals(self):
         candidates = glob.glob(self.pattern)
+        candidates.extend(glob.glob(self.pattern + self.etag_suffix))
 
         current = self._link(self._current_link)
         last = self._link(self._last_link)
         if current == self.current:
             # keep last+current
             self.remove(candidates, current)
             self.remove(candidates, last)
+            if current is not None:
+                self.remove(candidates, current + self.etag_suffix)
```

Comment:
wouldn't this remove something like current.etag?

Reply:
`self.remove()` removes `current.etag` from the list of things that should be cleaned up.
So in my understanding, the reverse is happening here.